extensions/amp-script/0.1/test/integration/test-amp-script.js

```diff
@@ -46,15 +48,36 @@ describe.configure().skipSinglePass().run('amp-script', function() {
     it('should say "hello world"', function*() {
       yield poll('<amp-script> to be hydrated',
           () => element.classList.contains('i-amphtml-hydrated'));
+      const impl = element.implementation_;
 
       // Give event listeners in hydration a moment to attach.
       yield browser.wait(100);
 
+      sandbox.stub(impl.userActivation_, 'isActive').callsFake(() => true);
```

Comment:
Is it possible to avoid referencing ivars and stubbing in integration tests? 

I wonder why the `click()` below doesn't trigger `isActive() == true`. Can also try dispatching `click` event.

Reply:
In this types of tests, `browser.click()` emulates a click by creating the event programmatically. The result of that is `event.isTrusted == false`, which is exactly what the `isTrusted` for. I can redo `UserActivationTracker` and skip using the `isTrusted` check since we control all events in AMP, but that'd still leave the non-active click test.

Reply:
To add to this, I started work on the e2e tests, which is a new integration testing framework. But connecting all dots there would still take time and I'd still need to test a non-active case.